packages/flutter_tools/lib/src/project.dart

```diff
@@ -395,39 +407,52 @@ class AndroidProject {
         || Cache.instance.isOlderThanToolsStamp(_ephemeralDirectory);
   }
 
-  Future<void> makeHostAppEditable() async {
+  Future<int> makeHostAppEditable({Map<String, dynamic> templateContext}) async {
     assert(isModule);
     if (_editableHostAppDirectory.existsSync())
       throwToolExit('Android host app is already editable. To start fresh, delete the android/ folder.');
-    _regenerateLibrary();
-    _overwriteFromTemplate(fs.path.join('module', 'android', 'host_app_common'), _editableHostAppDirectory);
-    _overwriteFromTemplate(fs.path.join('module', 'android', 'host_app_editable'), _editableHostAppDirectory);
-    _overwriteFromTemplate(fs.path.join('module', 'android', 'gradle'), _editableHostAppDirectory);
+
+    print('Editable host app directory: ${_editableHostAppDirectory.path}');
+    print('Template context: $templateContext');
+    int fileCount = 0;
+    fileCount += await _regenerateLibrary(templateContext: templateContext);
+//    fileCount += _overwriteFromTemplate(fs.path.join('module', 'android'), _editableHostAppDirectory, templateContext: templateContext, printStatusWhenWriting: true);
+    fileCount += await _overwriteFromTemplate(fs.path.join('module', 'android', 'host_app_common'), _editableHostAppDirectory, templateContext: templateContext, printStatusWhenWriting: true);
+    print('Writing language level android dir...');
+    fileCount += await _overwriteFromTemplate(fs.path.join('module', 'android', 'host_app_editable'), _editableHostAppDirectory, templateContext: templateContext, printStatusWhenWriting: true);
+    print('Done writing language level android dir.');
+    fileCount += await _overwriteFromTemplate(fs.path.join('module', 'android', 'gradle'), _editableHostAppDirectory, templateContext: templateContext, printStatusWhenWriting: true);
     gradle.injectGradleWrapper(_editableHostAppDirectory);
     gradle.writeLocalProperties(_editableHostAppDirectory.childFile('local.properties'));
     await injectPlugins(parent);
+
+    return fileCount;
   }
 
   File get localPropertiesFile => _flutterLibGradleRoot.childFile('local.properties');
 
   Directory get pluginRegistrantHost => _flutterLibGradleRoot.childDirectory(isModule ? 'Flutter' : 'app');
```

Comment:
References to `isModule` have to be re-worked.  In this world, every project will be a module.